packages/extension-browser/src/content-script/connector.ts

```diff
@@ -0,0 +1,192 @@
+import * as url from 'url';
+
+import { Engine } from 'hint';
+import { getContentTypeData, getType } from 'hint/dist/src/lib/utils/content-type';
+import {
+    ConnectorOptionsConfig,
+    HttpHeaders,
+    IAsyncHTMLDocument,
+    IAsyncHTMLElement,
+    IConnector,
+    FetchEnd,
+    NetworkData
+} from 'hint/dist/src/lib/types';
+
+import { Events } from '../shared/types';
+import { AsyncWindow, AsyncHTMLDocument, AsyncHTMLElement } from './web-async-html';
+import browser from '../shared/browser';
+
+export default class WebExtensionConnector implements IConnector {
+    private _document = new AsyncHTMLDocument(document);
+    private _window = new AsyncWindow(this._document);
+    private _engine: Engine;
+    private _options: ConnectorOptionsConfig;
+
+    public constructor(engine: Engine, options?: ConnectorOptionsConfig) {
+        this._engine = engine;
+        this._options = options || {};
+
+        if (!this._options.waitFor) {
+            this._options.waitFor = 1000;
+        }
+
+        browser.runtime.onMessage.addListener(async (events: Events) => {
+            if (events.fetchEnd) {
+                await this.notifyFetch(events.fetchEnd);
+            }
+            if (events.fetchStart) {
+                await this._engine.emitAsync('fetch::start', events.fetchStart);
+            }
+            // TODO: Trigger 'fetch::start::target'.
```

Comment:
Maybe the first message is the `target`? What we do in other connectors is to process the messages queued up first and then the rest and assume the first one is the target. This can be part of a separate PR BTW.